infrastructure/src/main/java/org/corfudb/infrastructure/log/FileSystemAgent.java

```diff
@@ -156,42 +148,86 @@ private void checkLimits() {
-        private PartitionAttribute partitionAttribute;
+        private volatile PartitionAttribute partitionAttribute;
+
+        // Path of the log partition, for example /config
         private final Path logPartition;
         private final FileSystemConfig config;
 
+        // A single thread scheduler that has a single instance of execution at any given time.
+        private static final ScheduledExecutorService scheduler =
+                Executors.newSingleThreadScheduledExecutor();
+
         public PartitionAgent(FileSystemConfig config) {
             this.config = config;
-            logPartition = Paths.get(config.logDir.getRoot().toString(), config.logDir.subpath(0, 1).toString());
-            partitionAttribute = setPartitionAttribute();
-            //we need a scheduler
+            // Joins root directory with its first sub path to get log partition.
+            // For example, "/" + "config"
+            logPartition = Paths.get(config.logDir.getRoot().toString(),
+                    config.logDir.subpath(0, 1).toString());
+            initializeScheduler();
+            setPartitionAttribute();
+        }
+
+        /**
+         * Resets PartitionAttribute's fields every {@link PartitionAttribute#UPDATE_INTERVAL}
+         * seconds after the previous set task is completed.
+         */
+        private void initializeScheduler(){
+            scheduler.scheduleWithFixedDelay(
+                    this::setPartitionAttribute, NO_DELAY, UPDATE_INTERVAL, SECONDS
+            );
         }
 
-        private PartitionAttribute setPartitionAttribute() {
+        /**
+         * Sets PartitionAttribute's fields with the values from log file and the log partition.
+         */
+        private void setPartitionAttribute() {
+            log.info("setPartitionAttribute: fetching PartitionAttribute.");
             try {
+                // Log path to check if it is in readOnly mode
                 File logDirectoryFile = config.logDir.toFile();
-                FileStore fileStore = Files.getFileStore(config.logDir);
+
+                // Partition of the log to fetch total and available space, and check readOnly
+                FileStore fileStore = Files.getFileStore(logPartition);
                 partitionAttribute = new PartitionAttribute(
                         fileStore.isReadOnly() || !logDirectoryFile.canWrite(),
                         fileStore.getUsableSpace(),
                         fileStore.getTotalSpace()
                 );
+                log.info("setPartitionAttribute: fetched PartitionAttribute successfully. " +
+                        "{}", partitionAttribute);
             } catch (IOException e) {
-                log.error("setPartitionStats: Error while getting PartitionStats", e);
+                log.error("setPartitionAttribute: Error while fetching PartitionAttributes. " +
+                        "Reinitializing the scheduler.", e);
+                initializeScheduler();
```

Comment:
I don't think this is needed; Since the exception is caught, the scheduler will still continue to execute this method periodically.